src/algebra/m231.rs

```diff
@@ -229,6 +230,17 @@ impl PartialEq<u32> for Mod231 {
     }
 }
 
+/// Try to solve the equation x^2 = a in our modular arithmetic
+/// Since MODULUS = 3 (mod 4), the solution, if exists, is x = a^((MODULUS+1)/4)
+pub fn try_sqrt(a: Mod231) -> Option<Mod231> {
+    let x = num_traits::pow::pow(a, MAGIC229);
```

Comment:
I was wondering, could we use [checked_pow] instead perhaps?

[checked_pow]: https://doc.rust-lang.org/std/primitive.u32.html#method.checked_pow

Reply:
no, we don't want to check for overflow, as we are using modular arithmetic